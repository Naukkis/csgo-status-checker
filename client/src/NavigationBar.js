import React from 'react';
import { NavItem, Nav, Navbar, NavDropdown, MenuItem, Modal } from 'react-bootstrap';
import Login from './Login';

class NavigationBar extends React.Component {
	constructor(props) {
		super(props);
		this.state = {showLogin: false};
		this.handleLogin = this.handleLogin.bind(this);
		this.closeLogin = this.closeLogin.bind(this);
	}

	handleLogin() {
		this.setState({showLogin: true});
	}

	closeLogin() {
		this.setState({showLogin: false});
	}

	render() {
		return (
			<Navbar inverse collapseOnSelect>
			    <Navbar.Header>
			      <Navbar.Brand>
			        <a href="#">CS:GO Status Checker</a>
			      </Navbar.Brand>
			      <Navbar.Toggle />
			    </Navbar.Header>
			    <Navbar.Collapse>
			      <Nav>
			        <NavItem eventKey={1} href="#">Link</NavItem>
			        <NavItem eventKey={2} href="#">Link</NavItem>    
			      </Nav>
			      <Nav pullRight>
				      <NavItem eventKey={1} onClick={this.handleLogin}>Login</NavItem>
				      <NavDropdown eventKey={3} title="Account" id="basic-nav-dropdown">
				          <MenuItem eventKey={3.1}>Action</MenuItem>
				          <MenuItem eventKey={3.2}>Another action</MenuItem>
				          <MenuItem eventKey={3.3}>Something else here</MenuItem>
				          <MenuItem divider />
				          <MenuItem eventKey={3.3}>Separated link</MenuItem>
			          </NavDropdown>			       
			      </Nav>
			    </Navbar.Collapse>
			  
			<Modal
				show={this.state.showLogin}
				onHide={this.closeLogin}
			>
			<Modal.Header closeButton>
				<Modal.Title>Login</Modal.Title>
			</Modal.Header>
			<Login closeModal={this.closeLogin}/>
			</Modal>
			</Navbar>
		)
	}
}

export default NavigationBar;