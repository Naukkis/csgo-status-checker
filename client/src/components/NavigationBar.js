import React from 'react';


class NavigationBar extends React.Component {

  render() {
    return (
      <ul className="navBar">
        <li><a href="http://localhost:3001/auth/steam">Login</a></li>
        <li><a href="/matches">Matches</a></li>
      </ul>
    );
  }
}

export default NavigationBar;
