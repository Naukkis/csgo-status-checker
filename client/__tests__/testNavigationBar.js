import React from 'react';
import renderer from 'react-test-renderer';
require('jest-localstorage-mock');
import NavigationBar from '../src/components/NavigationBar';

beforeEach(() => {
  // values stored in tests will also be available in other tests unless you run
  localStorage.clear();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<NavigationBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

localStorage.setItem('userID', 'lollolloo');

it('renders correctly with userID', () => {
  const tree = renderer
    .create(<NavigationBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});