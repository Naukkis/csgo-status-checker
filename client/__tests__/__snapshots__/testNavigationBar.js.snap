// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="navBar"
>
  <li>
    <a
      href="/"
    >
      Status Checker
    </a>
  </li>
  <li>
    <a
      href="/matches"
    >
      Matches
    </a>
  </li>
  <li
    style={
      Object {
        "float": "right",
      }
    }
  >
    <a
      href="/logout"
    >
      Log out
    </a>
  </li>
</ul>
`;

exports[`renders correctly with userID 1`] = `
<ul
  className="navBar"
>
  <li>
    <a
      href="/"
    >
      Status Checker
    </a>
  </li>
  <li>
    <a
      href="/matches"
    >
      Matches
    </a>
  </li>
  <li
    style={
      Object {
        "float": "right",
      }
    }
  >
    <a
      href="/logout"
    >
      Log out
    </a>
  </li>
</ul>
`;
